### PR TITLE
Modify trips hook

### PR DESCRIPTION
**Pull request by jaeivan**
#75

### CODE REVIEW

**Comment by jaeivan** [comment]
endpoint or functions not tested

**Comment by jaeivan** [comment]
BUG: req.body is defined but req.body.title is undefined

File: utility/hooks/trip.tsx [hunk 1]
```diff
@@ -250,6 +260,30 @@ export function TripProvider({
   // TODO: Allow a user to delete an availabillity widget for the trip
   async function deleteAvailabillityWidget() {}
 
+  async function createEvent(event: Event, callback: (response: Response) => void) {
```
Comment (Victoria-Williamson) [inline utility/hooks/trip.tsx:263]: need to store event in trip useState value

File: utility/hooks/trip.tsx [hunk 2]
```diff
@@ -250,6 +260,30 @@ export function TripProvider({
...
+
+    if (response.ok) {
+      callback({ isSuccess: response.ok, result: response.json() });
+
+    } else {
+      callback({ isSuccess: response.ok, errorMessage: await response.text() });
+    }
+  }
+
+  async function modifyTrip(trip: Trip, callback: (response: Response) => void) {
```
Comment (Victoria-Williamson) [inline utility/hooks/trip.tsx:275]: I don't think trip needs to be passed in, since we should have access to it in this hook

**Review by Victoria-Williamson — CHANGES_REQUESTED** [review]

File: pages/api/trip/[tripID]/[[...params]].ts [hunk 3]
```diff
@@ -70,9 +72,22 @@ export default async function handler(
               });
               break;
             }
+            case 'modify': {
+              if (req.body.title !== undefined) {
+                await updateDoc(docRef, {
```
Comment (Victoria-Williamson) [inline pages/api/trip/[tripID]/[[...params]].ts:77]: title isnt a value for the trip

File: pages/api/trip/[tripID]/[[...params]].ts [hunk 4]
```diff
@@ -70,9 +72,22 @@ export default async function handler(
...
+                  title: req.body.title,
+                });
+              }
+              if (req.body.destination !== undefined) {
```
Comment (Victoria-Williamson) [inline pages/api/trip/[tripID]/[[...params]].ts:81]: should also be updating duration, but both can be updated in the same updateDoc() function

File: utility/hooks/trip.tsx [hunk 5]
```diff
@@ -250,6 +267,36 @@ export function TripProvider({
...
 
+  async function createEvent(event: Event, callback: (response: Response) => void) {
+    const options = createFetchRequestOptions(JSON.stringify(trip), 'POST');
+    const response = await fetch(`${API_URL}/trip/${trip.uid}/event`, options);
+
+    if (response.ok) {
+      callback({ isSuccess: response.ok, result: response.json() });
+      setTrip({
+        ...trip,
+        joinableEvents: addEventToList(trip.joinableEvents, event),
```
Comment (Victoria-Williamson) [inline utility/hooks/trip.tsx:278]: ``` Array.from(addEventToList(trip.joinableEvents, event)) ```

File: utility/hooks/trip.tsx [hunk 6]
```diff
@@ -250,6 +267,36 @@ export function TripProvider({
...
+
+  async function modifyTrip(details: TripDetails, callback: (response: Response) => void) {
+    const options = createFetchRequestOptions(JSON.stringify(details), 'PUT');
+    const response = await fetch(`${API_URL}/trip/${trip.uid}/modify`, options);
+
+    if (response.ok) {
+      callback({ isSuccess: response.ok, result: response.json() });
+      setTrip({
+        ...trip,
+        destination: details.destination,
```
Comment (Victoria-Williamson) [inline utility/hooks/trip.tsx:293]: need to update duration too

**Review by Victoria-Williamson — CHANGES_REQUESTED** [review]

**Review by Victoria-Williamson — APPROVED** [review]

**Pull request merged**